sros2/test/sros2/commands/security/verbs/test_generate_policy.py

```diff
@@ -106,11 +103,6 @@ def test_generate_policy_topics(
         pub_sub_node_namespace,
         pub_sub_node_enclave
     ):
-        assert self.wait_for(expected_topics=[
-            expand_topic_name('~/pub', pub_sub_node_name, pub_sub_node_namespace),
-            expand_topic_name('~/sub', pub_sub_node_name, pub_sub_node_namespace),
-        ])
```

Comment:
So I understand that this patch is remove `wait_for` on topics in favor of an unconditional 4 second wait.  However, this is additionally losing the check that ensures that those topics exist (which as I understand is a necessary, but not sufficient, precondition).  Should we check that the topics exist before moving on with the rest of the test?  Similar question below for the services.

Reply:
> So I understand that this patch is remove wait_for on topics in favor of an unconditional 4 second wait.

That's right. 

> However, this is additionally losing the check that ensures that those topics exist (which as I understand is a necessary, but not sufficient, precondition)

These checks are useful when both test and `ros2` CLI use the daemon, as it speeds up the initial wait. When test and `ros2` CLI use their own node, it's useless (the fact that one discovered a topic or service does not imply the other has).

Thinking about this again, something's fishy. Flaky tests were often the ones using the daemon. Now I wonder if there's a connection with https://github.com/ros2/ros2cli/pull/620 I missed. I'll address that and see if it makes a difference here.

Reply:
Alright, I was indeed looking at the wrong place. The test implicitly assumes that topic or service discovery implies node discovery. That's not true. #260 fixes that by also waiting for the nodes, but only when a daemon is in use (it's pointless when not).